### PR TITLE
Use the `inspect` module to implement `Print_Function`

### PR DESCRIPTION
**Pull request by eric-wieser**
This will be less fragile, and is also shorter.

It's likely there'll be some minor latex output difference (extra blank lines etc), hence the draft status.

### CODE REVIEW

File: galgebra/printer.py [hunk 1]
```diff
@@ -1083,30 +1083,19 @@ def LatexFormat(Fmode=True, Dmode=True, ipy=False):
...
     return
 
-prog_str = ''
 off_mode = False
 
 
 def Get_Program(off=False):
-    global prog_str, off_mode
+    global off_mode
     off_mode = off
```
Comment (eric-wieser) [inline galgebra/printer.py:1091]: This function is now pretty useless - I'll deprecate it in a follow-up I think.

**Comment by eric-wieser** [comment]
Going to ignore the lgtm warning, imports are a mess in this file anyway, it will be easier to clean them all up later.

File: galgebra/printer.py (hunk 1 above)
Comment (utensil) [reply galgebra/printer.py:1091]: @abrombo used for displaying the program and the output in the same LaTeX file

Comment (eric-wieser) [reply galgebra/printer.py:1091]: That's what `Print_Function` is for. All this function does is say "make Print_Function do nothing when I call it".

Comment (utensil) [reply galgebra/printer.py:1091]: Oh

Comment (utensil) [reply galgebra/printer.py:1091]: I always confuse these two

**Pull request merged**